### ISSUE TITLE
Wrong media type for Opus audio

### ISSUE THREAD

**Issue by sewe** [mattgarrish/epub-revision#2516]
The [current draft](https://www.w3.org/TR/2023/CRD-epub-33-20230104/) lists `audio/opus` as media type for "OPUS audio using OGG container" and references [RFC 7845](https://www.rfc-editor.org/rfc/rfc7845) ("Ogg Encapsulation for the Opus Audio Codec").

While RFC 7845 is the correct reference and indeed defines how Opus audio is encapsulated inside an Ogg container, it does **not** define `audio/opus` as a media type for it. Instead, [Section 9 of RFC 7845](https://www.rfc-editor.org/rfc/rfc7845#section-9) recommends `audio/ogg`:

> An "Ogg Opus file" consists of one or more sequentially multiplexed
>  segments, each containing exactly one Ogg Opus stream.  The
> RECOMMENDED mime-type for Ogg Opus files is "audio/ogg".

Moreover, it even gives the following example:

> If more specificity is desired, one MAY indicate the presence of Opus
 > streams using the codecs parameter defined in [[RFC6381](https://www.rfc-editor.org/rfc/rfc6381)] and
>  [[RFC5334](https://www.rfc-editor.org/rfc/rfc5334)], e.g.,
>
>     audio/ogg; codecs=opus
>
> for an Ogg Opus file.

The RFC which does define `audio/opus` is [RFC 7587](https://www.rfc-editor.org/rfc/rfc7587.html) ("RTP Payload Format for the Opus Speech and Audio Codec"), but AFAIK this media type is for the RTP payload format, which is not Ogg encapsulated and not what one would typically find in an `.opus` **file**.

In other words, [Section 3.2 of the current draft](https://www.w3.org/TR/2023/CRD-epub-33-20230104/#sec-core-media-types) ("Core media types") lists the wrong media type for Opus audio.

**Comment by iherman** [comment]
(Disclaimer: I am not an audio expert...)

Thanks for your comment. I have looked at the [authoritative IETF list](https://www.iana.org/assignments/media-types/media-types.xhtml#audio) and I have found two entries:

- ogg: audio/ogg, ref rfc5334, rfc7845
- opus: audio/opus, ref rfc7587

Ie, there is indeed a discrepancy. Would the issue be solved if the EPUB document referred to RFC7587 instead of RFC7845?

I must admit I do not understand, at this point, the difference between ogg and opus. _Adding_ the ogg media type to the EPUB spec would be a bit too late at this point (but I am happy to record it as an issue to deal with later, if it is necessary).

Thanks for your comment, @sewe

**Comment by sewe** [comment]
(Disclaimer: I am not a RTP expert...)

> Would the issue be solved if the EPUB document referred to RFC7587 instead of RFC7845?

No, I don't think so.

The problem is that the two media types describe two **different** formats:

- `audio/ogg; codecs=opus` describes what one would customariy call an "Opus file" and what is most certainly what you want in an EPUB container.
- `audio/opus` describes the over-the-wire format used during a RTP transmission.

Have a look at the [actual registration for `audio/opus`](https://www.iana.org/assignments/media-types/audio/opus) and you see lots of parameters referring to "sender" and "receiver", i.e., an ongoing RTP transmission.

Inside an EPUB file, you really want to use the "OPUS audio using OGG container" **file** format and the spec is okay in that regard. You just use the wrong media type to refer to what you want.

> _Adding_ the ogg media type to the EPUB spec would be a bit too late at this point (but I am happy to record it as an issue to deal with later, if it is necessary).

Not sure where exactly you see the problem. It's not about adding support for a new media type, but just fixing the the spec:

Hence, the table in section 3.2 should really read

audio/ogg; codecs=opus | [[rfc7845](https://www.w3.org/TR/2023/CRD-epub-33-20230104/#bib-rfc7845)] | OPUS audio using OGG container
-- | -- | --

Effectively, this means that `/manifest/item/@media-type` should **not** use `audio/opus`, but rather 
`audio/ogg; codecs=opus`.

If you are worried that such a change might break already deployed implementations, then maybe also support `audio/opus` as a legacy "alias", but noting that this still denotes the "OPUS audio using OGG container" format described in RFC 7845 rather than the RTP over-the-wire format described in RFC 7587. (The latter RFC should at most be an informative reference, as it's not really "used" by the spec.)

**Comment by mattgarrish** [comment]
> If you are worried that such a change might break already deployed implementations

Opus is new to the 3.3 revision, so I'm not sure there's much real risk of that.

The only concern I have is that we haven't defined parameters for core media types before. We reference RFC 2046 for them, which doesn't include the codecs parameter, so we'd have to update the definition to include [RFC 4281](https://datatracker.ietf.org/doc/html/rfc4281).

**Comment by mattgarrish** [comment]
> which doesn't include the codecs parameter

This is maybe a non-issue after looking through 2046 some more. It defines that there can be parameters, so although `codecs` is defined separately that's not really relevant for the purposes of our spec. We just need valid media types.
